### PR TITLE
Fix 'reverse' naming collision

### LINKED ISSUE #392: reverse bug?
image

but:

{% assign my_array = "abc,123" | split: "," %}
{{ my_array | reverse | join: "," }}

Output:
image

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #392

### CODE REVIEW

**Comment by hishamco** [comment]
I will move the filter into `MiscFilter` then should be ready to merge

**Comment by sebastienros** [comment]
Got some new knowledge, and `reverse` doesn't work on strings as we thought. It is really an Array filter, and when applied on strings it returns an array of chard. Check this sample https://shopify.github.io/liquid/filters/reverse/ and the correct behavior in LiquidJs: https://liquidjs.com/playground.html#e3sgImZvbyIgfCByZXZlcnNlIH19,e30=

The current implementation is wrong, it will probably surprise people using Fluid currently.

**Comment by hishamco** [comment]
You are right it works for Arrays only, but when I did for String I reference some Liquid docs, let me check please

**Comment by hishamco** [comment]
Seb seems I miss from where I reference this, I think we have two choices here:

1. Let `reverse` applied for arrays only
2. Let `reverse` applied for arrays and strings

I think we already introduce new filters or functionality that not included in https://shopify.github.io/liquid, correct me if I'm wrong

**Comment by sebastienros** [comment]
It works for strings too, it returns an Array of chars, not a reversed string.

**Comment by sebastienros** [comment]
But it can be kept in the arrays filters class.

**Comment by hishamco** [comment]
@sebastienros can we merge this?

File: Fluid/Filters/ArrayFilters.cs [hunk 1]
```diff
@@ -103,6 +103,45 @@ public static async ValueTask<FluidValue> Map(FluidValue input, FilterArguments
...
+                    return StringValue.Empty;
+                }
+                else
+                {
+                    String reversedString;
+                    var valueAsArray = value.ToCharArray();
+#if NETSTANDARD2_0
+                    Array.Reverse(valueAsArray);
+
+                    reversedString = new String(valueAsArray);
```
Comment (sebastienros) [inline Fluid/Filters/ArrayFilters.cs:126]: Wrong. I mentioned that in another comment, this filter when applied on strings actually returns an array of chars, not a string.

**Comment by sebastienros** [comment]
This comment points to examples: https://github.com/sebastienros/fluid/pull/393#issuecomment-962644292
`'abc' | reverse` returns `['c', 'b', 'a']`

File: Fluid/Filters/ArrayFilters.cs (hunk 1 above)
Comment (hishamco) [reply Fluid/Filters/ArrayFilters.cs:126]: Oops seems I forgot to push the local changes

**Comment by hishamco** [comment]
While we don't have `FluidChar` I used `FluidString` instead, we could probably introduce it - if it's necessary - while `FluidString` may used it internally

**Comment by sebastienros** [comment]
@hishamco can you try to add this test?

```
{% assign my_array = "abc,123" | split: "," %}
{{ my_array | reverse | join: "," }}
```

I am not sure this PR is actually fixing the issue, though it is fixing the reverse on strings.

**Comment by sebastienros** [comment]
I did and it's working

**Comment by hishamco** [comment]
Seems I late to respond due my Internet connection unstable, thanks for testing that case too

**Pull request merged**